charts/jaeger-operator/templates/service.yaml

```diff
@@ -11,7 +11,10 @@ spec:
     port: 8383
     protocol: TCP
     targetPort: 8383
+{{- if and (eq .Values.service.type "NodePort") (.Values.service.nodePort) }}
```

Comment:
The current version will end with an error if you set `nodePort`.

you can try it like this
```
{{ if (and (eq .Values.service.type "NodePort") (not (empty .Values.service.nodePort))) }}
    nodePort: {{.Values.service.nodePort}}
{{- end }}
```

Reply:
This is the same conditional statement as used in the existing `jaeger/templates/query-svc` helm chart: https://github.com/jaegertracing/helm-charts/blob/157d2d5c5d43dfc9bde975ef5b0baf99f42c2c1f/charts/jaeger/templates/query-svc.yaml#L19

Reply:
```diff
-{{ .Values.query.service.nodePort }}
+{{ .Values.service.nodePort }}
```

Reply:
The naming of the variables aside, the logic of the conditional statement is the same and shouldn't result in an error as it is logically equivalent to the one in your suggestion.

Reply:
OK, change the path to the variable to the correct path